engines/pink/cel_decoder.cpp

```diff
@@ -0,0 +1,225 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#include "common/stream.h"
+
+#include "graphics/surface.h"
+
+#include "pink/cel_decoder.h"
+
+namespace Pink {
+
+bool CelDecoder::loadStream(Common::SeekableReadStream *stream) {
+	close();
+
+	/* uint32 frameSize = */ stream->readUint32LE();
+	uint16 frameType = stream->readUint16LE();
+
+	// Check FLC magic number
+	if (frameType != 0xAF12) {
+		warning("FlicDecoder::loadStream(): attempted to load non-FLC data (type = 0x%04X)", frameType);
+		return false;
+	}
+
+	uint16 frameCount = stream->readUint16LE();
+	uint16 width = stream->readUint16LE();
+	uint16 height = stream->readUint16LE();
+	uint16 colorDepth = stream->readUint16LE();
+	if (colorDepth != 8) {
+		warning("FlicDecoder::loadStream(): attempted to load an FLC with a palette of color depth %d. Only 8-bit color palettes are supported", colorDepth);
+		return false;
+	}
+
+	addTrack(new CelVideoTrack(stream, frameCount, width, height));
+	return true;
+}
+
+
+uint16 CelDecoder::getTransparentColourIndex() {
+	CelVideoTrack *track = (CelVideoTrack *)getTrack(0);
+	if (!track)
+		return 0;
+	return track->getTransparentColourIndex();
+}
+
+const Graphics::Surface *CelDecoder::getCurrentFrame() {
+	CelVideoTrack *track = (CelVideoTrack *)getTrack(0);
+	if (!track)
+		return 0;
+	return track->getCurrentFrame();
+}
+
+Common::Point CelDecoder::getCenter() {
+	CelVideoTrack *track = (CelVideoTrack *)getTrack(0);
+	if (!track)
+		return Common::Point(0, 0);
+	return track->getCenter();
+}
+
+void CelDecoder::skipFrame() {
+	CelVideoTrack *track = (CelVideoTrack *)getTrack(0);
+	track->skipFrame();
+}
+
+void CelDecoder::setEndOfTrack() {
+	CelVideoTrack *track = (CelVideoTrack *)getTrack(0);
+	track->setEndOfTrack();
+}
+
+CelDecoder::CelVideoTrack::CelVideoTrack(Common::SeekableReadStream *stream, uint16 frameCount, uint16 width, uint16 height, bool skipHeader)
+		: FlicVideoTrack(stream, frameCount, width, height, 1), _center(0, 0), _transparentColourIndex(0) {
+	readHeader();
+}
+
+#define PREFIX_TYPE 0xF100
+#define CEL_DATA 3
```

Comment:
It appears these two macros are only used by the following method without being `#undef`'d again thus polluting global namespace.